### ISSUE TITLE
make this work with jitpack

### ISSUE THREAD

**Issue by adriancole** [line/armeria#1732]
jitpack allows you to pull jars on-demand using tag or commit based version identifiers. @bsideup helped me troubleshoot why I couldn't do this in armeria.

https://jitpack.io/com/github/anuraaga/armeria/unary-grpc-service-armeria-0.4.0.Final-g0685edc-1693/build.log

```
    Cannot publish a release version without a GPG key; missing signatory
```

Armeria's build seems to enforce GPG even on install. Can this be changed without impacting you?

**Comment by trustin** [comment]
Would it be enough to add a build flag like `-Pnosign`?

**Comment by adriancole** [comment]
if signing is needed by default in the install goal, jitpack would need a custom config file to use the profile to avoid it https://jitpack.io/docs/BUILDING/#custom-commands

**Comment by trustin** [comment]
I guess we can autodetect if we are running in JitPack by checking `JITPACK` environment variable. Let me send a fix soon.

**Comment by trustin** [comment]
The fix is coming: https://github.com/line/gradle-scripts/pull/55

**Comment by trustin** [comment]
Hopefully fixed via 028b023572594688c9b4c51c89b26de777e4f24f
Please reopen if it didn't work in the next release.
